t/50mruby-prometheus.t

```diff
@@ -47,19 +47,17 @@ sub parse_output {
             fail("invalid typeline: $typeline");
             return;
         }
-        if ($valueline =~ /^$name\{version="(.*)"\} (.*)$/) {
-            $version = $1;
-            $value = $2;
-        } else {
-            fail("invalid valueline: $valueline");
-            return;
+        # there can be multiple valuelines per typeline
+        foreach my $valueline (@lines) {
+            last unless ($valueline =~ /^$name\{version="(.*)"\} (.*)$/);
+            push(@ret, +{
+                name => $name,
+                type => $type,
+                version => $1,
+                value => $2,
+            });
+            shift @lines;
```

Comment:
I think these modifications don't work. First of all, this test is currently passing because no `h2o_evloop_latency_nanosec` value lines are emitted using the above configuration. Add
```
duration-stats: ON
num-threads: 2 # to ensure multiple valuelines are emitted
```
to it then you will see this test starts to fail. There would be several issues and also several ways to fix, but jumping to the conclusion, the patch below works well and also adds an additional check for valueline attributes.

```
diff --git a/t/50mruby-prometheus.t b/t/50mruby-prometheus.t
index 5acbc3d8a..5dbd342c2 100644
--- a/t/50mruby-prometheus.t
+++ b/t/50mruby-prometheus.t
@@ -10,6 +10,8 @@ plan skip_all => 'mruby support is off'
 
 subtest "basic" => sub {
     my $server = spawn_h2o(<< 'EOT');
+duration-stats: ON
+num-threads: 2
 hosts:
   default:
     paths:
@@ -48,15 +50,25 @@ sub parse_output {
             return;
         }
         # there can be multiple valuelines per typeline
-        foreach my $valueline (@lines) {
-            last unless ($valueline =~ /^$name\{version="(.*)"\} (.*)$/);
+        while (@lines && $lines[0] !~ /^#/) {
+            my $valueline = splice(@lines, 0, 1);
+            unless ($valueline =~ /^$name\{version="(.*?)"(?:, (.+))?\} (.*)$/) {
+                fail("invalid valueline: $valueline");
+            }
+            my %attrs = ();
+            if (my $other_attrs = $2) {
+                for my $attr (split(', ', $other_attrs)) {
+                    $attr =~ /(.+)="(.+)"/ or fail("invalid valueline attribute: $attr");
+                    $attrs{$1} = $2;
+                }
+            }
             push(@ret, +{
                 name => $name,
                 type => $type,
                 version => $1,
-                value => $2,
+                value => $3,
+                %attrs,
             });
-            shift @lines;
         }
     }
     return \@ret;

```